azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -36,20 +37,24 @@
 TRANSPORT_PRV_FILE_NAME = "TransportPrivate.pem"
 
 
+class GoalStateRetrievalMode:
+    Default = 0
+    Limited = 1
+    Full = 2
+
+
 class GoalState(object):
     #
     # Some modules (e.g. telemetry) require an up-to-date container ID. We update this variable each time we
     # fetch the goal state.
     #
     ContainerID = "00000000-0000-0000-0000-000000000000"
+    _HttpFailedIndicator = "[HTTP Failed]"
 
-    def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
+    def __init__(self, wire_client, ext_config_retriever, retrieval_mode=GoalStateRetrievalMode.Default):
```

Comment:
The retrieval_mode parameter is very important. Let's remove the default value from the parameter and force the caller to explicitly say what mode they want.

Reply:
made it required